### ISSUE TITLE
Can use 'readonly' as type-name but cannot make use of it

### ISSUE THREAD

**Issue by jrieken** [microsoft/TypeScript#42760]
# Bug Report

* ts 4.1.3
* take the snippet below
* you get unexpected compile errors on the property declaration

```ts
type readonly = 'readonly';
class Fun {
	readonly readonly: readonly = 'readonly';
}
```

**Comment by jrieken** [comment]
This issue might be one for the bin but my goal is write a long sausage of `readonly`, like

```

type readonly = readonly readonly[]

class Fun {
	readonly readonly: readonly = []
}
```

**Comment by RyanCavanaugh** [comment]
A noble endeavor, to be sure.

The `type readonly = ''` declaration should be made illegal.
